### DISCUSSION TITLE
Any plans on having a UI like Blockly for writing expressions?

### DISCUSSION THREAD

**Discussion by rchougule** [antonmedv/expr#701] [General]
Hello,

We are planning to use expr for our RuleEngine system being built to evaluate Rules on some transactions. Though, I was wondering if there is any web-ui possible on writing the expressions in a much more simpler manner.

For e.g., https://developers.google.com/blockly helps in converting blocks to code. These blocks are built by dragging components around. Is it possible to connect these two in some manner? Or a similar tool where analysts or any non-developer can have UI blocks and convert it to an expression.

Thanks!

**Comment by antonmedv** [comment]
Hello,

I’m actually working on a web UI that would make writing expressions more accessible, especially for non-developers. I would love to understand more about what you’re specifically looking for in such a tool. Do you have any particular requirements or features in mind, like the types of blocks or the complexity of expressions you need to support?

It might be helpful to schedule a meeting so we can discuss your needs in more detail and see how we can align our efforts. Let me know if that works for you.

Best regards.

**Reply by rchougule** [reply]
Apologies for the delayed reply here.
Thanks for responding so eagerly. Let me collate some requirements and get back.
